xbmc/settings/windows/GUIControlSettings.cpp

```diff
@@ -543,7 +544,7 @@ bool CGUIControlButtonSetting::OnClick()
     }
     else if (controlFormat == "path")
       SetValid(GetPath((CSettingPath *)m_pSetting));
-    else if (controlFormat == "action")
+    else if (controlFormat == "action" || controlFormat == "infoaction")
```

Comment:
That doesn't answer my question. Do you have a usecase where you add a setting with the "infoaction" control format and register for the `OnSettingAction()` callback which is triggered when the user activates the label representing the setting? If not it would IMO be better to add a "label" control format with its own `CGUIControlLabelSetting` implementation (from which `CGUIControlButtonSetting` could be derived).